src/cargo/core/compiler/job_queue.rs

```diff
@@ -245,8 +257,12 @@ impl<'a> JobQueue<'a> {
                 }
                 Message::Finish(key, result) => {
                     info!("end: {:?}", key);
-                    self.active -= 1;
-                    if self.active > 0 {
+
+                    // self.active.remove_item(&key); // <- switch to this when stabilized.
+                    if let Some(pos) = self.active.iter().position(|k| *k == key) {
```

Comment:
This if let some looks suspicious to me. It should never fail, right? If that’s the case, let’s replace it with expect? If we later make a bug somewhere around here, expect would be much easier to debug.

Reply:
Will do :)